### DISCUSSION TITLE
How to disable the default domain name completely?

### DISCUSSION THREAD

**Discussion by hydrogenoxide-user** [vercel/vercel#6525] [Help]
I encountered a problem. After I bound my own domain name, such as mysite.com, there are still additional domain names that can be accessed in the project, such as porject-git-user.vercel.app. How can I disable them completely, because it is vulnerable to malicious attacks!

**Comment by leerob** [comment]
You can redirect those other domains back to your canonical URL, if you would like. Could you share more on how having multiple URLs makes it vulnerable for attacks?

**Reply by hydrogenoxide-user** [reply]
I deployed the waline comment system on the top of vercel. Yesterday, it was written by malicious data. I resisted it with cloudflare, but the attacker bypassed the firewall from the default .app domain name. I didn't see any options about .app domain name in the menu of vercel. It was so annoying that I didn't need it at all. How do I delete or redirect these. .app domain names?

**Comment by azvyae** [comment]
This is for people that coming into this discussion and want to solve the redirection. Try to add `middleware.js` on root of your project. Then add conditional checking on the `middleware.js` like this:

```javascript
import { NextResponse } from "next/server";
export function middleware(request) {
  if (!request.nextUrl.origin.includes("vercel.app")) {
    return NextResponse.next();
  }
  return NextResponse.redirect("https://example.com", 301);
}
```

That way will redirect the vercel.app generated domain to intended domain, permanently, and will works for future deployments
